cs3/gateway/v1beta1/gateway_api.proto

```diff
@@ -534,6 +538,26 @@ message ListAuthProvidersResponse {
   repeated string types = 3;
 }
 
+message ListExistingReceivedSharesResponse {
+  // REQUIRED.
+  // The response status.
+  cs3.rpc.v1beta1.Status status = 1;
+  // OPTIONAL.
+  // Opaque information.
+  cs3.types.v1beta1.Opaque opaque = 2;
+  // REQUIRED.
+  // The underlying shares as returned by the collaboration service.
+  repeated cs3.sharing.collaboration.v1beta1.ReceivedShare shares = 3;
```

Comment:
I prefer having a new message encapsulating the ReceivedShare and the ResourceInfo to avoid the clients having to do the mapping themselves.

Reply:
Makes sense, something like `repeated SharedResourceInfo shares = 3` with `SharedResourceInfo` being the encapsulating message (that should maybe live in `cs3.sharing.collaboration.v1beta1`?)

Reply:
(Answering myself: no, this is a gateway resource)